sys/types.h might not always include sys/sysmacros.h for major/minor/makedev

--- a/mksquashfs.c
+++ b/mksquashfs.c
@@ -59,6 +59,7 @@
 #else
 #include <endian.h>
 #include <sys/sysinfo.h>
+#include <sys/sysmacros.h>
 #endif
 
 #include "squashfs_fs.h"
--- a/unsquashfs.c
+++ b/unsquashfs.c
@@ -38,6 +38,10 @@
 #include <limits.h>
 #include <ctype.h>
 
+#ifdef linux
+#include <sys/sysmacros.h>
+#endif
+
 struct cache *fragment_cache, *data_cache;
 struct queue *to_reader, *to_inflate, *to_writer, *from_writer;
 pthread_t *thread, *inflator_thread;
